Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SaveVulcanGSS.py

```diff
@@ -0,0 +1,462 @@
+        :param parm_file_name:
+        :return:
+        """
+        # check that workspace shall be point data
+        if output_workspace.isHistogramData():
+            raise RuntimeError('Output workspace shall be point data at this stage.')
+
+        # construct the headers
+        vulcan_gsas_header = self.create_vulcan_gsas_header(output_workspace, gsas_file_name, ipts_number,
+                                                            parm_file_name)
+
+        vulcan_bank_headers = list()
+        for ws_index in range(output_workspace.getNumberHistograms()):
+            bank_id = output_workspace.getSpectrum(ws_index).getSpectrumNo()
+            bank_header = self.create_bank_header(bank_id, output_workspace.readX(ws_index))
+            vulcan_bank_headers.append(bank_header)
+        # END-F
+
+        # Save
+        try:
+            api.SaveGSS(InputWorkspace=output_workspace, Filename=gsas_file_name, SplitFiles=False, Append=False,
+                        Format="SLOG", MultiplyByBinWidth=False,
+                        ExtendedHeader=False,
+                        UserSpecifiedGSASHeader=vulcan_gsas_header,
+                        UserSpecifiedBankHeader=vulcan_bank_headers,
+                        UseSpectrumNumberAsBankID=True,
+                        SLOGXYEPrecision=[1, 1, 2])
+        except RuntimeError as run_err:
+            raise RuntimeError('Failed to call SaveGSS() due to {0}'.format(run_err))
+
+        return
+
+    def create_vulcan_gsas_header(self, workspace, gsas_file_name, ipts, parm_file_name):
+        """
+        create specific GSAS header required by VULCAN team/VDRIVE
+        :param workspace:
+        :param gsas_file_name:
+        :param ipts:
+        :param parm_file_name:
+        :return:
+        """
+        # Get necessary information including title, run start, duration and etc.
+        title = workspace.getTitle()
+
+        # Get run object for sample log information
+        run = workspace.getRun()
+
+        # Get information on start/stop
+        if run.hasProperty("run_start") and run.hasProperty("duration"):
+            # have run start and duration information
+            run_start = run.getProperty("run_start").value
+            duration = float(run.getProperty("duration").value)
+
+            # separate second and sub-seconds
+            run_start_seconds = run_start.split(".")[0]
+            run_start_sub_seconds = run_start.split(".")[1]
+            self.log().warning('Run start {0} is split to {1} and {2}'.format(run_start, run_start_seconds,
+                                                                              run_start_sub_seconds))
+
+            # property run_start and duration exist
+            utctime = datetime.strptime(run_start_seconds, '%Y-%m-%dT%H:%M:%S')
+            time0 = datetime.strptime("1990-01-01T0:0:0", '%Y-%m-%dT%H:%M:%S')
+
+            # convert from string version of time to total nano second
```

Comment:
The better solution is `np.datetime64`. It can be created from those time stamps and converted into nanoseconds from the start using
```python
utctime = np.datetime64(run.getProperty("run_start").value)
time0 = n.datetime64("1990-01-01T0:0:0")
total_nanosecond_start = int((utctime-time0)/np.timedelta64(1, 'ns'))
```
Then a bunch of your other code goes away.